### ISSUE TITLE
Should dhtnode package *require* dhtnode-common?

### ISSUE THREAD

**Issue by gavin-norman-sociomantic** [sociomantic-tsunami/dhtnode#11]
At the moment, dhtnode-common is only [_recommended_](https://github.com/sociomantic-tsunami/dhtnode/blob/master/pkg/dhtnode.pkg#L12) (not required). This seems wrong to me. If you install dhtnode, you always need the contents of dhtnode-common (the upstart scripts).

**Comment by nemanja-boric-sociomantic** [comment]
See the documentation on this topic: https://github.com/sociomantic/backend/blob/master/doc/deployment/packaging/stage2.md#dealing-with-common-files

```
    # The common files are recommended, as (even when rare) someone could want
    # to install just the binaries for a quick test without any upstart or
    # logrotate configuration. By default Ubuntu will install recommended
    # packages
```

So, this is to allow for the binary package to be installed without common files, but as a unseen complication comes that the assumption about default installing recommended packages is wrong. I don't think allowing for this possibility is worth it, though. cc @leandro-lucarella-sociomantic

**Comment by gavin-norman-sociomantic** [comment]
Yeah, I think the default should be to always install everything needed for a live installation.

**Comment by nemanja-boric-sociomantic** [comment]
Yes, but I think the reasoning here is not about the default, but for actually _allowing_ this combination (just dhtnode without dhtnode-common package). As far as I know, there's no way to install just the package without its strong dependencies through `apt`,

**Comment by nemanja-boric-sociomantic** [comment]
There seems to be a workaround: https://askubuntu.com/a/80794

**Comment by leandro-lucarella-sociomantic** [comment]
I don't feel terribly married to the idea of keeping it as recommended. I think it makes sense, but if you think it's too inconvenient, we could change it. Although for live systems it shouldn't be a problem as tools like puppet can take care of making sure all needed packages are installed.
